### ISSUE TITLE
Founder status on Hostmasks should be allowed, but there must exist at least one account based founder

### ISSUE THREAD

**Issue by sebastiannielsen** [atheme/atheme#656]
If I try this on my channel in a IRC server:

`/MSG ChanServ ACCESS #mychannel ADD *!*@123.123.123.123 FOUNDER` (where 123.123.123.123 is my real IP)

It will complain:
`[23:43:43] <ChanServ> You may not set founder status on a hostmask.`

I do understand the rationale of it, because if the IP changes, and the only founder on channel is a hostmask, you could lose your channel.

But I think the check could be improved, so if atheme is set on the server side to allow multiple founders on a channel, it could allow hostmask-based founders, provided theres atleast 1 account-based founder, but atheme will still prevent deleting a account-based founder if that the only account-based founder on channel left.

So basically, there MUST be atleast one account-based founder, but after that, hostmask-based founders should be allowed.

**Comment by grawity** [comment]
I always thought the rationale is the opposite – that hostmask ACEs are insecure because it's sometimes easy to obtain someone else's hostname.

(Been there, done that.)

**Comment by sebastiannielsen** [comment]
grawity: Yes, if a ISP have bad checks on who can obtain PTR's you could be in trouble if you use such a hostmask (with a domain name) and a malicious person were able to trick his ISP to set a PTR for his IP for who he don't own the upstream domain. Thats why you should use literal IPs (or their anonymous ounterparts, like XXXX-xxx.xxx.xxxxxx.IP on network who use IP obfuscation) when configuring hostmask access controls.

**Comment by jlu5** [comment]
What additional things do you want a hostmask founder to be able to do that isn't allowed by assigning all flags except +F?

**Comment by KoraggKnightWolf** [comment]
An example might be chanserv/unban as this appears limited to those who are logged into an account

**Comment by alyx** [comment]
What benefits are there to allowing this; **why** would hostmask based founder status be an improvement over the current way Atheme handles ACLs? (No, "atheme shouldn't restrict what you can do" doesn't count)

**Comment by kaniini** [comment]
Hostmask ACL entries are an insecure misfeature that exist in Atheme solely for luddites who thought refusing to register with NickServ was an expression of their wokeness.  Allowing hostmask ACL entries to have founder rights will lead to channel takeovers.  If anything, hostmask ACLs in their entirety, should be removed, in the year of our lord 2019.

**Comment by sebastiannielsen** [comment]
@kaniini I disagree with your opinion.
ACL entires are actually more secure, provided they are used in the right way AND the corresponding ircd have set up securely (with FCrDNS verification in both directions).
Of course, setting ACLs like SomeNick!whatever@* or *!*@123.123.123.*
of course leads to channel takeovers, as removing your lock from your front door or putting a shitty lock there, will lead to burgulars.

But used right, it can be even better than password authentication.

Also on servers which do NOT perform FCrDNS verification in both direction (ergo both look up PTR record, and also verify the domain in PTR record resolves back to the IP via a full traverse through DNS), you could easily set up a bouncer on a host that allows custom PTR with no verification, and either set the PTR to an authorized IP-adress (and thus cause the ircd PTR lookup to fool Atheme into thinking the person connecting have the right channel access IP) or you could plain simply enter the right PTR if the access entry are set up with domain name instead of IP.

So both of these must be true for host ACLs to be secure, and if both of these are true, host ACL become very secure, as you can't "bypass" or "hack" a host-based ACL. Passwords can be hacked or stolen, but to be able to authorize as a specific host, you must physically connect from that host.

Using host-based ACLs, apart from allowing non-registred users to get channel access, but its also a very secure way to for example enforce secure channel access to users without them being able to change that, for example locking their AOP or SOP access to a specific physical location only.

I know that some reckless users might set *!*@* as FOUNDER and get their channel taken over, but its their fault then. Should a door manufacturer make it impossible to remove the keylock from the door, just because some owner might remove the lock and leave the door unlocked? But still, you could make use of the feature to install a biometric lock (compared to a good hostmask that ends in like @123.123.123.123 a literal IP, on a well configured ircd that verifies FCrDNS in both directions)

**Comment by kaniini** [comment]
For those cases, you should lock down accounts to allow login from only specified hostmasks.  Exposing channels to hostmasks ACLs with real power is like giving idiots a loaded shotgun pointed straight at their foot.  My opinion is based on reality, I'm the one who wrote the code allowing hostmasks to be used to begin with.  I know why I wrote it.

**Comment by aaronmdjones** [comment]
> Passwords can be hacked or stolen, but to be able to authorize as a specific host, you must physically connect from that host.

If you accept that passwords can sometimes be stolen, then you must also accept that it is sometimes possible for people to connect from a given host; after all, you could steal the password used to access it. The corollary (e.g. the host is using publickey-based logins instead of passwords) also applies to Atheme (May I introduce you to `/NS CERT ADD` and `/NS SET NOPASSWORD ON` ?), so again it is no better than having (and correctly using) a services account.

**Comment by ilbelkyr** [comment]
Consensus on this appears to be reasonably clearly against supporting this, and I'd tend to agree; closing.
